### ISSUE TITLE
Indentation of assignment RHS after one-line comment

### ISSUE THREAD

**Issue by davidBar-On** [rust-lang/rustfmt#4502]
**Output**
Current expected formatting output from [tests/target/issue-3851.rs](https://github.com/rust-lang/rustfmt/blob/2d6a968d52b13b75fc3799f50020a34efd480a28/tests/target/issue-3851.rs#L7-L11)
```rust
    let after =
    // after_comment
    if true {
        1.0
    };
```

**Expected output**
To be compatible with other formatting, **the desired output may be with the `if` statement indented to the right**:
```rust
    let after =
    // after_comment
        if true {
            1.0
        };
```

For example, following is **current formatting output of similar expression**:
```rust
    let aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa =
        if cccccccccccccccccccccccccccccccccccccccccccccccccc {
            1.0
        };
```

**Comment by calebcartwright** [comment]
Generally agreed, though note that the comment should be correctly indented as well:

```rust
let after =
        // after_comment
        if true {
            1.0
        };
```

**Comment by davidBar-On** [comment]
Following are some test cases from #4518 with issues that seem to be related to this issue.  **Is it worth trying to fix these issues?**

1. `if /*w*/ 5 /*x*/ == /*y*/ 6 /*z*/ {}` is formatted as follows (I am not sure what should be the right format)
```rust
    if
    /*w*/
    5 /*x*/ == /*y*/ 6
    /*z*/
    {}
```

2. `let x /* yyy */ = /* x as */ 0f64 as i32;` is formatted as follows.  Note general indentation and the one space instead of tab in front of `0f64` (not sure if it is this issue, cast `as` issue or both):
```rust
    /* yyy */
    let x =
    /* x as */
     0f64 as i32;
```

3. The following code:
```rust
    if a == b // x
    {}
```
is formatted as:
```rust
    if a == b
    // x
    {}
```

**Comment by calebcartwright** [comment]
> is formatted as:

Is that a description current behavior, or a suggested expected/target formatted output? If the latter, is the indentation shown above correct? Here's how that's showing in the GH UI

![image](https://user-images.githubusercontent.com/13042488/101438924-0a7db000-38d9-11eb-8c82-e050e60a3b05.png)

**Comment by davidBar-On** [comment]
This is the **current** format output.  An extra space is added before the `777`.  The two problems described in this issue are supposed to be **fixed by PR #4550**.  As I noted in the PR, I through that these two issues (indentation after one-line comment and extra space in the following line) are related, but it turned out that they are not.  However, since the fix for each is simple (assuming what I did is correct), I submitted both fixes in one PR.
